### PR TITLE
Variod testbench

### PR DESCRIPTION
**Pull request by bomilkar**
Tools to run variod in a controlled environment and on machines which have no sensor H/W.
sensord_mock : supplies NMEA sentences to variod like sensord would normally do.
my_netcat : implements "nc -l" on platforms which don't have that functionality.

More details in the README.txt

### CODE REVIEW

**Comment by bomilkar** [comment]
This PR has just the testbench related stuff in it.
It isolated in the "testbench" directory.

**Comment by linuxianer99** [comment]
@bomilkar : Same here. Please rebase the delete the revert commits.

**Review by bomilkar — COMMENTED** [review]
Done

**Comment by bomilkar** [comment]
@linuxianer99 is there anything I can do to make this move?

File: variod.c [hunk 1]
```diff
@@ -312,52 +311,9 @@ int main(int argc, char *argv[])
...
-
-				} else {
-					fprintf(stderr, "send failed\n");
 				}
 			}
 
-			// use specific data from received messge locally
-			// strtok will gobble up the content of client_message
-			next_sentence = strtok(client_message,sentence_delimiter);
-
```
Comment (kedder) [inline variod.c:324]: I believe this diff to `variod.c` does not belong to this PR. Please get rid of it (unless I'm mistaken).

**Review by kedder — CHANGES_REQUESTED** [review]

**Comment by kedder** [comment]
@linuxianer99, I'm pretty sure @bomilkar will learn how to squash commits eventually. Meanwhile, there is an option to "squash and merge" in github on the green merge button. It might not preserve the commit authorship, but who cares, right?

I believe this is pretty useful tool for development. I was able to run it on my laptop and even heard the vario sound. After changes to `variod.c` is removed from this PR, I think it is good to go!

**Comment by bomilkar** [comment]
Rebase was the request, not squash.

File: variod.c (hunk 1 above)
Comment (bomilkar) [reply variod.c:324]: This was from the first method to make it handle multiple NMEA sentences from one recv().
In the meantime I came up with a more elegant, efficient and robust way to achieve the same goal.

**Comment by bomilkar** [comment]
I'll squash all commits into one if that's what it takes to make it move on.
Please advise, @linuxianer99 .

**Comment by bomilkar** [comment]
One more point, maybe it's not clear: I added this testbench capability as an initial commit not as a change of what existed before.
On the first attempt I had it munged up with other stuff changing variod. That was a bad idea on my part, so I reverted all the variod related commits and what is left is the initial commit for testbench.
If you review the changes (of all commits) the you see: it only affects files in the subdir testbench.
I hope that makes it clear.

**Comment by DanD222** [comment]
Hi @bomilkar - I think the ultimate aim is to review just one commit instead of 17 that there currently are, so what I would recommend is performing an interactive rebase which squashes all the relevant commits into one, and then doing a force-push to overwrite the 17 commits in your PR with your one commit. You can also chose “skip” during the interactive rebase to leave out any unrelated commits rather than reverting them. 

You can then repeat this process in case you make further adjustments to the code in the PR, etc.

**Comment by kedder** [comment]
@bomilkar If you look at "Commits" tab in this PR, you'll see 17 commits, and most of them have nothing to do with this feature.  If this PR is merged like it is now, all these commits will go to the "mainline" commit history and stay there forever, confusing anyone who wants to dig into the history. To keep histoy clean, it is considered a good practice to split a feature to small number of self-contained changes with a clear purpose. To my eye, this PR doesn't deserve more than 1 commit, since it s an initial implementation of a feature.

It is a usual practice to commit a lot while you are developing a feature, but "clean up" the history before requesting to pull into the upstream, so that your "work-in-progress" commits would not contaminate the history for everyone. As @DanD222 mentioned, the common tool for that is "git interactive rebase". You can also see how the diff looks like before pushing the branch with `git diff` to avoid unwanted changes to squeeze into the PR.

**Comment by bomilkar** [comment]
OK, understand.
Since it's a new feature (not a change to existing code) I will close this PR, delete the branch and start with a new PR.
In this particular case, there is no history.
